sdk/core/azure-core/inc/azure/core/internal/tracing/tracing_impl.hpp

```diff
@@ -0,0 +1,291 @@
+   *    - std::span<const int32_t>                          *** Not Implemented
+   *    - std::span<const int64_t>                          *** Not Implemented
+   *    - std::span<const uint32_t>                         *** Not Implemented
+   *    - std::span<const double>                           *** Not Implemented
+   *    - std::span<std::string/std::string_view>           *** Not Implemented
+   *    - uint64_t (not fully supported)                    *** Not Implemented
+   *    - std::span<uint64_t> (not fully supported)         *** Not Implemented
+   *    - std::span<const uint8_t> (not fully supported)    *** Not Implemented.
+   *
+   */
+  class AttributeSet {
+  public:
+    /**
+     * @brief Adds a Boolean attribute to the attribute set.
+     *
+     * @param attributeName Name of attribute to add.
+     * @param value Value of attribute.
+     */
+    virtual void AddAttribute(std::string const& attributeName, bool value) = 0;
+    /**
+     * @brief Adds a 32bit integer attribute to the attribute set.
+     *
+     * @param attributeName Name of attribute to add.
+     * @param value Value of attribute.
+     */
+    virtual void AddAttribute(std::string const& attributeName, int32_t value) = 0;
+    /**
+     * @brief Adds a 64bit integer attribute to the attribute set.
+     *
+     * @param attributeName Name of attribute to add.
+     * @param value Value of attribute.
+     */
+    virtual void AddAttribute(std::string const& attributeName, int64_t value) = 0;
+    /**
+     * @brief Adds an unsigned 64bit integer attribute to the attribute set.
+     *
+     * @param attributeName Name of attribute to add.
+     * @param value Value of attribute.
+     */
+    virtual void AddAttribute(std::string const& attributeName, uint64_t value) = 0;
+    /**
+     * @brief Adds a 64bit floating point attribute to the attribute set.
+     *
+     * @param attributeName Name of attribute to add.
+     * @param value Value of attribute.
+     */
+    virtual void AddAttribute(std::string const& attributeName, double value) = 0;
+    /**
+     * @brief Adds a C style string attribute to the attribute set.
+     *
+     * @param attributeName Name of attribute to add.
+     * @param value Value of attribute.
+     */
+    virtual void AddAttribute(std::string const& attributeName, const char* value) = 0;
+    /**
+     * @brief Adds a C++ string attribute to the attribute set.
+     *
+     * @param attributeName Name of attribute to add.
+     * @param value Value of attribute.
+     */
+    virtual void AddAttribute(std::string const& attributeName, std::string const& value) = 0;
+
+    /**
+     * @brief destroys an AttributeSet - virtual destructor to enable base class users to
+     * destroy derived classes.
+     */
+    virtual ~AttributeSet() = default;
+  };
+
+  /** @brief The Type of Span.
+   */
+  class SpanKind final : public Azure::Core::_internal::ExtendableEnumeration<SpanKind> {
```

Comment:
DOes this really needs to be an extensible enum instead of the regular one? We use the extensible ones for the stuff that gets relayed over network, gets received from REST API calls etc.

Reply:
The same question for other extensible enums.

Reply:
Strictly speaking, no. Changing.